source/unified-test-format/unified-test-format.rst

```diff
@@ -2635,22 +2635,34 @@ sections.
 Terminating Open Transactions
 `````````````````````````````
 
-Open transactions can cause tests to block indiscriminately. Test runners SHOULD
-terminate all open transactions at the start of a test suite and after each
-failed test by killing all sessions in the cluster. Using the internal
-MongoClient, execute the ``killAllSessions`` command on either the primary or,
-if connected to a sharded cluster, all mongos servers.
+Open transactions can cause tests to block indiscriminately. When connected to
+MongoDB 3.6 or later, test runners SHOULD terminate all open transactions at the
+start of a test suite and after each failed test by killing all sessions in the
+cluster. Using the internal MongoClient, execute the ``killAllSessions`` command
+on either the primary or, if connected to a sharded cluster, all mongos servers.
 
 For example::
 
     db.adminCommand({
       killAllSessions: []
     });
 
-The test runner MAY ignore any command failure with error Interrupted(11601) to
-work around `SERVER-38335`_.
+The test runner MAY ignore the following command failures:
+
+- Interrupted(11601) to work around `SERVER-38335`_.
+- Unauthorized(13) to work around `SERVER-54216`_.
+- CommandNotFound(59) if the command is executed on a pre-3.6 server
 
 .. _SERVER-38335: https://jira.mongodb.org/browse/SERVER-38335
+.. _SERVER-54216: https://jira.mongodb.org/browse/SERVER-54216
+
+Note that Atlas, by design, does not allow database users to kill sessions
+belonging to other users. This makes it impossible to guarantee that an existing
+transaction will not block test execution. To work around this, test runners
+SHOULD either ignore Unauthorized(13) command failures or avoid calling
+``killAllSessions`` altogether when connected to Atlas (e.g. by detecting
+``mongodb.net`` in the hostname or allowing the test runner to be configured
+externally).
```

Comment:
The goal of this paragraph is to advise test runners not to call the command on Atlas or at least ignore a failure if they do. I'm intentionally not being prescriptive about how the test runner decides to do that, since I imagine some folks might strongly prefer once mechanism over another. IMO, detecting the hostname is probably the most flexible solution.